pkg/client/query/cache/memory.go

```diff
@@ -10,116 +10,119 @@ import (
 )
 
 var (
-	_ client.QueryCache[any]           = (*InMemoryCache[any])(nil)
-	_ client.HistoricalQueryCache[any] = (*InMemoryCache[any])(nil)
+	_ client.QueryCache[any]           = (*inMemoryCache[any])(nil)
+	_ client.HistoricalQueryCache[any] = (*inMemoryCache[any])(nil)
 
 	DefaultQueryCacheConfig = queryCacheConfig{
-		EvictionPolicy: FirstInFirstOut,
+		evictionPolicy: FirstInFirstOut,
 		// TODO_MAINNET(@bryanchriswhite): Consider how we can "guarantee" good
-		// alignment between the TTL and the block production rate.
-		TTL: time.Minute,
+		// alignment between the TTL and the block production rate,
+		// by accessing onchain block times directly.
+		ttl: time.Minute,
 	}
 )
 
-// InMemoryCache provides a concurrency-safe in-memory cache implementation with
+// inMemoryCache provides a concurrency-safe in-memory cache implementation with
 // optional historical value support.
-type InMemoryCache[T any] struct {
+type inMemoryCache[T any] struct {
 	config       queryCacheConfig
 	latestHeight atomic.Int64
 
-	itemsMu sync.RWMutex
-	// items type depends on historical mode:
-	// | historical mode |  type                          |
-	// | --------------- | ------------------------------ |
-	// | false           | map[string]cacheItem[T]        |
-	// | true            | map[string]cacheItemHistory[T] |
-	items map[string]any
+	// valuesMu is used to protect values AND valueHistories from concurrent access.
+	valuesMu sync.RWMutex
+	// values holds the cached values in non-historical mode.
+	values map[string]cacheValue[T]
+	// valueHistories holds the cached historical values in historical mode.
+	valueHistories map[string]cacheValueHistory[T]
 }
 
-// cacheItem wraps cached values with a timestamp for later comparison against
+// cacheValue wraps cached values with a cachedAt for later comparison against
 // the configured TTL.
-type cacheItem[T any] struct {
-	value     T
-	timestamp time.Time
+type cacheValue[T any] struct {
+	value    T
+	cachedAt time.Time
 }
 
-// cacheItemHistory stores cachedItems by height and maintains a sorted list of
+// cacheValueHistory stores cachedItems by height and maintains a sorted list of
 // heights for which cached items exist. This list is sorted in descending order
 // to improve performance characteristics by positively correlating index with age.
-type cacheItemHistory[T any] struct {
+type cacheValueHistory[T any] struct {
 	// sortedDescHeights is a list of the heights for which values are cached.
 	// It is sorted in descending order.
 	sortedDescHeights []int64
-	itemsByHeight     map[int64]cacheItem[T]
+	heightMap         map[int64]cacheValue[T]
```

Comment:
I was considering `heightCacheValueMap`, but `heightMap` just doesn't feel like enough.
```suggestion
	heightValueMap         map[int64]cacheValue[T]
```

Reply:
Continuing the direction of https://github.com/pokt-network/poktroll/pull/994#discussion_r1890025275, I'm going with `heightVersionsMap` here.